### PR TITLE
ci: Speculative checks

### PR DESCRIPTION
**Pull request by mslipper**
Updates Mergify rules to support speculative checks. This will allow Mergify to speculatively run checks against multiple queued PRs at once, and reduce the maount of time we spend waiting for PRs to get merged in the queue.

Also removes the changeset nag, and updates the messaging around Mergify queue failures.

### CODE REVIEW

**Review by tynes — APPROVED** [review]

File: .github/mergify.yml [hunk 1]
```diff
@@ -89,23 +91,19 @@ pull_request_rules:
...
         message: |
           This PR is next in line to be merged, and will be merged as soon as checks pass.
-  - name: Notify author when merge queue failed
+  - name: Notify author on queue failure
     conditions:
-      - "check-failure=Queue: Embarked in merge train"
+      - 'check-failure=Queue: Embarked in merge train'
     actions:
       comment:
-        message: |
```
Comment (smartcontracts) [inline .github/mergify.yml:97]: Why remove this?

Comment (mslipper) [reply .github/mergify.yml:97]: It's replaced with another message that explains how to fix the problem.

Comment (smartcontracts) [reply .github/mergify.yml:97]: Oh sorry I meant why remove the "Nag changesets" message

Comment (trianglesphere) [reply .github/mergify.yml:97]: We already have the changesets reminder

Comment (smartcontracts) [reply .github/mergify.yml:97]: I see

**Comment by mslipper** [comment]
@mergifyio refresh

**Pull request merged**